### PR TITLE
dash/edit-mode-context

### PR DESCRIPTION
**Pull request by karolkolodziej**
Added context detection for custom HTML.

### CODE REVIEW

**Comment by stitot** [comment]
Looking good! Some issues:

**Toogle is deactivated**
When editMode is enabled by default (like in demo) the toggle button is inactive. Activating it turn editMode off. 

**Context menu jumps down**
1. Click hamburger menu
2. Enable editMode
3. Disable editMode

Observe that the context menu jumps down (tested in chrome)

**Fields miss label** (see attachment)

Skjermbilde 2024-06-08 kl  13 24 51

**Icon on mouseover**
I want us to consider that the edit icon (the cogwheel) should also appears on mouseover, together with the blue border. The extra click needed doesn't seem necessary?

**Edit pane always to the left**
The edit pane is never triggered to the right, even when it covers the chart I want to edit to the left.

**Comment by karolkolodziej** [comment]
> Fields miss label (see attachment)

Unrelated to the changes, I reported this separately. 

> Icon on mouseover

It behaves the same as the layout version- IMO we shouldn't change that now. At least not within the PR.

**Review by vazonik — APPROVED** [review]

File: ts/Dashboards/EditMode/SidebarPopup.ts [hunk 1]
```diff
@@ -272,8 +274,12 @@ class SidebarPopup extends BaseForm {
         }
 
         // Remove highlight from the row.
-        if (editMode.editCellContext && editMode.editCellContext.row) {
-            editMode.editCellContext.row.setHighlight(true);
+        if (
+            editMode.editCellContext &&
+            editMode.editCellContext instanceof Cell &&
+            editMode.editCellContext.row
```
Comment (sebastianbochan) [inline ts/Dashboards/EditMode/SidebarPopup.ts:280]: `editMode.editCellContext instanceof Cell &&
            editMode.editCellContext.row` it should be enough.

File: ts/Dashboards/EditMode/SidebarPopup.ts [hunk 2]
```diff
@@ -445,14 +451,19 @@ class SidebarPopup extends BaseForm {
             editMode.setEditOverlay(true);
         }
 
-        if (editCellContext && editCellContext.row) {
+        if (
+            editCellContext &&
+            editCellContext instanceof Cell &&
+            editCellContext.row
```
Comment (sebastianbochan) [inline ts/Dashboards/EditMode/SidebarPopup.ts:457]: The same as above

File: docs/dashboards/edit-mode.md [hunk 3]
```diff
@@ -1,8 +1,11 @@
 Edit mode
 ===
 
-Edit mode is a mode in which the user can change the appearance of the dashboard through the User Interface (UI).
-To be able to use Dashboards with layout system and editmode you first have to load the `layout` module.
+Edit mode is a mode in which the user can change the appearance of the dashboard through the User Interface (UI).  
+When dashboard is declared with custom HTML structure, the edit mode is limited to the component settings only.
```
Comment (jomar-honsi) [inline docs/dashboards/edit-mode.md:5]: the dashboard

**Review by jomar-honsi — APPROVED** [review]

**Review by jomar-honsi — APPROVED** [review]

**Review by stitot — CHANGES_REQUESTED** [review]
I meant it when I said "1000000000" ;-)

For this to actually reduce cases of overlap due to high z-index in customers layout I believe it must be much higher than 100.

**Comment by sebastianbochan** [comment]
The zIndex like `1000000000` does not make sense, because the maximum value is limited. I suggest setting the value 9999.

**Comment by stitot** [comment]
> The zIndex like `1000000000` does not make sense, because the maximum value is limited. I suggest setting the value 9999.

@karolkolodziej mentioned there was maybe a more elegant solution than this "brute force" one.

**Comment by stitot** [comment]
Have a look at e.g. http://localhost:3030/samples/#view/data-grid/v2/fixed-distribution

If I remove `width: 100% `from `table` it's still 100%, I guess because of the inline width we set on `thead`?

Possible to have default table behavior in these cases? That table is never wider than needed if no explicit width for the table is set via CSS?

**Review by sebastianbochan — APPROVED** [review]
Looks good to me.

**Review by stitot — COMMENTED** [review]
I would like to understand your reluctance to set this higher. Value for z-index is an integer, so can be −2147483648 to +2147483647 without breaking the rules. Could we at least add some 9's, so that the value is 9999999? The higher the better, as it will cover more cases. Your call.

**Pull request merged**